pkg/image/oci/registry_provider.go

```diff
@@ -38,12 +38,7 @@ func (p *RegistryImageProvider) Provide() (*image.Image, error) {
 		return nil, err
 	}
 
-	var options []name.Option
-	if p.registryOptions.InsecureUseHTTP {
-		options = append(options, name.Insecure)
-	}
-
-	ref, err := name.ParseReference(p.imageStr, options...)
+	ref, err := name.ParseReference(p.imageStr, prepareReferenceOptions(p.registryOptions)...)
```

Comment:
Because `p.registryOptions` is a pointer type, we should check to see if its value is `nil` at some point. In this situation, I could see two choices:

1. Check for its nilness here in the body of `Provide`. If you took this approach, you could simplify the signature of `prepareReferenceOptions` by having it accept a `image.RegistryOptions` (which can be relied on as not nil) instead of a `*image.RegistryOptions`.
2. Put the nilness check inside the body of `prepareReferenceOptions`.

The reason this matters is that `prepareReferenceOptions` has the expression `registryOptions.InsecureUseHTTP` which will assume that `registryOptions` **_is not nil_**, and if it actually is nil, the application will panic.

Reply:
Added `nil` check to `prepareReferenceOptions`: https://github.com/anchore/stereoscope/pull/76/files#diff-a48bf3359f783f78022ab9cd098b21e3c4c9262c2e6319f55a11bc0004cddc09R74